blockchair/src/endpoint/balance.ts

```diff
@@ -0,0 +1,137 @@
+import objectPath from 'object-path'
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { AdapterRequest, Config, Address, Account } from '@chainlink/types'
+import { DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { CoinType, ChainType, COINS } from '.'
+
+export const Name = 'balance'
+
+type AddressGroup = {
+  addresses: string[]
+  result: Address[]
+}
+
+type GroupedAddresses = {
+  [network: string]: AddressGroup
+}
+
+type ResponseWithResult = {
+  response: any
+  result: Account[]
+}
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+const WARNING_NO_OPERATION_TESTNET =
+  'No Operation: only testnet supported by blockchair adapter is btc'
+
+const group = (jobRunID: string, addresses: Address[]) => {
+  const output: GroupedAddresses = {}
+  for (const addr of addresses) {
+    if (!addr.address)
+      throw new AdapterError({
+        jobRunID,
+        message: `Addresses are required`,
+        statusCode: 400,
+      })
+    addr.coin = addr.coin || 'btc'
+    addr.chain = addr.chain || 'mainnet'
+    const { address, coin, chain } = addr
+    const key = `${coin}-${chain}`
+    output[key] = output[key] || { addresses: [], result: [] }
+    output[key].addresses.push(address)
+    output[key].result.push(addr)
+  }
+  return output
+}
+
+const getBalanceURI = (jobRunID: string, addresses: string[], chain: string, coin: string) => {
+  coin = Requester.toVendorName(coin, COINS)
+  if (!coin)
+    throw new AdapterError({
+      jobRunID,
+      message: `Unsupported coin parameter`,
+      statusCode: 500,
+    })
+  if (chain === 'testnet') {
+    if (coin === 'bitcoin') coin = 'bitcoin/testnet'
+    else
+      throw new AdapterError({
```

Comment:
We should not throw here in the first place. If we can't process an item, we add a warning to it and continue.